contracts/NewfiAdvisor.sol

```diff
@@ -173,25 +269,33 @@ contract NewfiAdvisor is Initializable, ReentrancyGuardUpgradeSafe, ProxyFactory
         address _advisor,
         uint256 _stableProportion,
         uint256 _volatileProportion
-    ) external payable {
+    ) external payable nonReentrant {
+        // Token Dispersal Logic 
+        // if msg.value > 0
+        // AggregatorInterface(fiatRef).latestAnswer().mul(msg.value) --> getting the usdc value of ether deposited
+        // stable pool token price dispersed to the investor = bal of usdc in advisor stable pool / total supply this will give price
+        // stableInvest * price is the tokens advisor gets which will be stable liquidity in struct
+        // volatile pool token price dispersed to the investor = bal of usdc in advisor stable pool (after converted from eth to usdc check 1st comment in this function)/ total supply this will give price
+        // volatilePoolAmount(after converted from eth to usdc check 1st comment in this function) * price is the tokens advisor gets which will be stable liquidity in struct
         require(
             _stableProportion + _volatileProportion == 100,
             "Need to invest 100% of funds"
         );
         Advisor storage advisor = advisorInfo[_advisor];
         IERC20 token = IERC20(_stablecoin);
-        uint256 stableInvest = _totalInvest * _stableProportion / 100;
-        uint256 volatileInvest = _totalInvest * _volatileProportion / 100;
-        uint256 ethAmount = msg.value.div(10 ** uint256(18));
+        uint256 stableInvest = (_totalInvest * _stableProportion) / 100;
+        uint256 volatileInvest = (_totalInvest * _volatileProportion) / 100;
+        // getting the total amount in volatile pool
+        uint256 volatilePoolAmount = volatileInvest.add(msg.value);
```

Comment:
if you add 1000 stable coins to 100000000 wei then you'll have a big number that doesn't mean anything. This is why they are split.